Riot/Modules/Room/RoomViewController.m

```diff
@@ -7721,6 +7721,15 @@ - (void)documentPickerPresenter:(MXKDocumentPickerPresenter *)presenter didPickD
 {
     self.documentPickerPresenter = nil;
     
+    // Check maxUploadSize accepted by the home server before trying to upload.
+    NSUInteger maxUploadFileSize = self.roomDataSource.mxSession.maxUploadSize;
+    NSDictionary *fileAttributes = [NSFileManager.defaultManager attributesOfItemAtPath:url.path error:nil];
+    if (fileAttributes && fileAttributes.fileSize > maxUploadFileSize) {
+        [self showAlertWithTitle:@"The file you want to upload is too big."
+                         message:[NSString stringWithFormat:@"\nThe file you want to upload is too big.\n\nIt mustn't weight more than %ldMB", maxUploadFileSize/(1024*1024)]];
+        return;
```

Comment:
This whole check is in the wrong place, it doesn't account for videos and images to be processed, meaning something that would previously send, will now be rejected. It should either happen after the respective processing has taken place, or only happen for files inside the `else if (fileUTI.isFile)`.